### PR TITLE
Add logistic variable transformation

### PR DESCRIPTION
**Pull request by marpulli**
Add logistic transformation so variables can be constrained to be within lower and upper bound

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: mxfusion/components/variables/var_trans.py [hunk 1]
```diff
@@ -84,3 +85,46 @@ def __init__(self):
...
+
+        :param out_var: Variable to be transformed.
+        :type out_var: mx.ndarray or mx.sym
+        :param F: Mode to run MxNet in.
+        :type F: mxnet.ndarray or mxnet.symbol
+        :param dtype: data type.
+        :type dtype: e.g. np.float32
+        """
+        F = get_default_MXNet_mode() if F is None else F
+        return F.log(F.clip(out_var - self._lower, 1e-10, np.inf) / F.clip(self._upper - out_var, 1e-10, np.inf))
```
Comment (meissnereric) [inline mxfusion/components/variables/var_trans.py]:
```suggestion
        return F.log(F.clip(out_var - self._lower, 1e-10, np.inf) / F.clip(self._upper - out_var, 1e-10, np.inf))
        
```

File: testing/core/var_trans_test.py [hunk 2]
```diff
@@ -38,3 +38,10 @@ def test_softplus_numerical(self, x, rtol, atol):
         npt.assert_allclose(mf_pos.asnumpy(), np_pos, rtol=rtol, atol=atol)
         npt.assert_allclose(mf_inv.asnumpy(), np_inv, rtol=rtol, atol=atol)
         npt.assert_allclose(mf_inv.asnumpy(), x.asnumpy(), rtol=rtol, atol=atol)
+
+    def test_logistic(self):
```
Comment (meissnereric) [inline testing/core/var_trans_test.py]: Can you add some parameterizations to this like in the test_softplus_numerical above? Fine to not compare against numpy, just to test it against a few values.

File: mxfusion/components/variables/var_trans.py [hunk 3]
```diff
@@ -84,3 +85,47 @@ def __init__(self):
...
+
+        :param var: Variable to be transformed.
+        :type var: mx.ndarray or mx.sym
+        :param F: Mode to run MxNet in.
+        :type F: mxnet.ndarray or mxnet.symbol
+        :param dtype: data type.
+        :type dtype: e.g. np.float32
+        """
+        F = get_default_MXNet_mode() if F is None else F
+        return self._lower + self._difference / (1. + F.exp(-var))
```
Comment (zhenwendai) [inline mxfusion/components/variables/var_trans.py]: Could you please use the activation operator of mxnet instead? (mxnet.ndarray.Activation(data, act_type='sigmoid)

This straight implementation has numerical issue when var goes too big.

File: mxfusion/components/variables/var_trans.py [hunk 4]
```diff
@@ -84,3 +85,47 @@ def __init__(self):
...
+
+        :param out_var: Variable to be transformed.
+        :type out_var: mx.ndarray or mx.sym
+        :param F: Mode to run MxNet in.
+        :type F: mxnet.ndarray or mxnet.symbol
+        :param dtype: data type.
+        :type dtype: e.g. np.float32
+        """
+        F = get_default_MXNet_mode() if F is None else F
+        return F.log(F.clip(out_var - self._lower, 1e-10, np.inf) / F.clip(self._upper - out_var, 1e-10, np.inf))
```
Comment (zhenwendai) [inline mxfusion/components/variables/var_trans.py]: Why not clip the out_var first like F.clip(out_var, self._lower, self._upper)?

**Comment by zhenwendai** [comment]
@marpulli Could you please update your branch? Thanks!

**Pull request merged**